File: May Challenge/(5) First Unique Character/FirstUniqueCharacter.py
'''
First Unique Character in a String

Solution
Given a string, find the first non-repeating character in it and return it's index. If it doesn't exist, return -1.

Examples:

s = "leetcode"
return 0.

s = "loveleetcode",
return 2.
Note: You may assume the string contain only lowercase letters.
'''

class Solution:
    def firstUniqChar(self, s: str) -> int:
        for x in range(len(s)):
            if s.count(s[x])> 1:
                continue
            else:
                return x
        return -1

s=Solution()
s.firstUniqChar("leetcode")

File: May Challenge/(5) First Unique Character/FirstUniqueCharacter.kt
class Solution {
    fun firstUniqChar(s: String): Int {
        val count = HashMap<Char, Int>()
        val n = s.length
        // build hash map : character and how often it appears
        for (i in 0 until n) {
            val c = s[i]
            count[c] = count.getOrDefault(c, 0) + 1
        }

        // find the index
        for (i in 0 until n) {
            if (count[s[i]] === 1) return i
        }
        return -1
    }
}
